### PR TITLE
Add rules for inset and top, left, right bottom classes

### PR DESCRIPTION
**Pull request by nwikstrand**
This adds rules for various inset classes and top, left, right, bottom.

rules support the following, value suffix can of course differ, bounding will be added later.

```
inset-auto
inset-full
inset-1/2
inset-y-1
inset-x-1
top-1
left-1
right-1
bottom-1
```

All credit to @BalbinaK  for setting this up 🙌

### CODE REVIEW

File: plugin/src/_rules/position.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    if (r != null && d in insetMap)
+        return insetMap[d].map(i => [i.slice(1), r]);
+}
+export const insets = [
+    [/^inset-(.+)$/, ([, v], ctx) => ({ inset: handleInsetValue(v, ctx) }),
+        {
+            autocomplete: [
+                '(inset-<directions>-$spacing',
+                'inset-(block|inline)-$spacing',
+                'inset-(bs|be|is|ie)-$spacing',
```
Comment (pearofducks) [inline plugin/src/_rules/position.js:17]: autocomplete needs cleaned up?

**Review by pearofducks — APPROVED** [review]
once autocomplete is cleaned up this lgtm

File: plugin/src/_rules/position.js (hunk 1 above)
Comment (nwikstrand) [reply plugin/src/_rules/position.js:17]: Most def, will fix

Comment (nwikstrand) [reply plugin/src/_rules/position.js]: [be1322a](https://github.com/warp-ds/engine/pull/3/commits/be1322a797a93133bf77f44fb7343d2f83452d9a)

This ok or am I missing something, brain is tired, please halp

**Review by Skadefryd — APPROVED** [review]
Looks good to me :)

File: plugin/src/_rules/position.js (hunk 1 above)
Comment (pearofducks) [reply plugin/src/_rules/position.js]: yep that's good!

**Comment by nwikstrand** [comment]
Gaaah, forgot to squash the fixup 🤦

File: plugin/src/_rules/position.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    const r = handleInsetValue(v, ctx);
+    if (r != null && d in insetMap)
+        return insetMap[d].map(i => [i.slice(1), r]);
+}
+export const insets = [
+    [/^inset-(.+)$/, ([, v], ctx) => ({ inset: handleInsetValue(v, ctx) }),
+        {
+            autocomplete: [
+                '(inset-$spacing',
+                '(inset-<direction>-$spacing',
```
Comment (BalbinaK) [inline plugin/src/_rules/position.js:16]: @pearofducks Should we also allow autocomplete with "full", "auto", "1/2", "1/4"?

Comment (pearofducks) [reply plugin/src/_rules/position.js:16]: It seems like we might already be including that here? At least the original preset-mini autocomplete code doesn't have any special consideration for those?

I think it's a great thing to note down somewhere (a 'test autocomplete' task?) or keep in our mind for when we do check that though!

**Pull request merged**